File: find_first_pallindromic_string_in_the_array.go
func firstPalindrome(words []string) string {
    for _, word := range words {
        flag := false

        for i, j := 0, len(word)-1; i < j; i, j = i+1, j-1 {
            if word[i] != word[j] {
                flag = true
                break
            }
        }
        if !flag {
            return word
        }
    }
    return ""
}


// Time Complexity : O(n*m)
// Space Complexity : O(1)

File: find_first_pallindromic_string_in_the_array.cpp
class Solution {
public:
    string firstPalindrome(vector<string>& words) {
        for(auto word: words) {
            string temp = word;
            reverse(word.begin(), word.end());
            if(word == temp) return word;
        }
        return "";
    }
};

// Time Complexity : O(n*m)
// Space Complexity : O(1)